lib/rubocop/cop/performance/caller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where `caller.first`
+      # can be replaced by `caller(1..1).first`.
+      #
+      # @example
+      #   # bad
+      #   caller.first
+      #
+      #   # good
+      #   caller(1..1).first
+      class Caller < Cop
+        MSG = 'Use `caller(1..1)` instead of `caller`.'.freeze
+        SCOPE_METHODS = %i(first []).freeze
+
+        def_node_matcher :caller_with_scope_method?, <<-PATTERN
+          (send (send nil :caller ...) ${#{SCOPE_METHODS.map(&:inspect).join(' ')}} ...)
+        PATTERN
+
+        def on_send(node)
+          return unless caller_with_scope_method?(node) && slow_caller?(node)
+          add_offense(node, :selector)
+        end
+
+        private
+
+        def slow_caller?(node)
+          arguments = node.receiver.arguments
+
+          arguments.empty? ||
+            (arguments.length == 1 && arguments[0].type == :int)
```

Comment:
Should this check `arguments[0].int_type?` instead of `arguments[0].type == :int`?